Userland/Libraries/LibJS/Console.cpp

```diff
@@ -511,24 +512,128 @@ ThrowCompletionOr<Value> ConsoleClient::logger(Console::LogLevel log_level, Mark
         return printer(log_level, move(first_as_vector));
     }
 
-    // 5. If first does not contain any format specifiers, perform Printer(logLevel, args).
-    if (!TRY(first.to_string(vm)).contains('%')) {
-        TRY(printer(log_level, args));
-    } else {
-        // 6. Otherwise, perform Printer(logLevel, Formatter(args)).
+    // 5. Otherwise, perform Printer(logLevel, Formatter(args)).
+    else {
         auto formatted = TRY(formatter(args));
         TRY(printer(log_level, formatted));
     }
 
-    // 7. Return undefined.
+    // 6. Return undefined.
     return js_undefined();
 }
 
 // 2.2. Formatter(args), https://console.spec.whatwg.org/#formatter
 ThrowCompletionOr<MarkedVector<Value>> ConsoleClient::formatter(MarkedVector<Value> const& args)
 {
-    // TODO: Actually implement formatting
-    return args;
+    auto& vm = m_console.vm();
+    auto& realm = *vm.current_realm();
+    MarkedVector<Value> result { vm.heap() };
```

Comment:
Can we defer this until step 7 please? no need to allocate here and then throw it away in a bunch of cases :)

Reply:
:+1: